src/lib/comms/sol-coap.c

```diff
@@ -130,6 +130,18 @@ struct outgoing {
 
 static bool on_can_write(void *data, struct sol_socket *s);
 
+/* This is an internal API and shouldn't be marked with SOL_API. */
+struct sol_socket *sol_coap_server_get_socket(const struct sol_coap_server *server);
+
+struct sol_socket *
+sol_coap_server_get_socket(const struct sol_coap_server *server)
```

Comment:
It's necessary to obtain the sol_socket_dtls to derivate the key from the initial handshake. (It's in sol-oic-security.c, part of this PR, which is not shown in the GitHub UI due to size limitations.)